queue/go/queue.go

```diff
@@ -0,0 +1,37 @@
+package main
+
+// Queue is an implementation of a queue
+// using a doubly-linked list.
+type Queue struct {
+	linkedList doublyLinkedList
+}
+
+// Enqueue adds an element to the back of the queue.
+func (q Queue) Enqueue(element interface{}) {
+	q.linkedList.AddLast(element)
+}
+
+// Dequeue removes the element at the front of the queue
+// and returns it. It returns an error if the queue is empty.
+func (q Queue) Dequeue() (interface{}, error) {
+	return q.linkedList.RemoveFirst()
+}
+
+// Peek returns the element at the front of the queue.
+// It returns an error if the queue is empty.
+func (q Queue) Peek() (interface{}, error) {
+	return q.linkedList.Peek()
+}
+
+type doublyLinkedList interface {
```

Comment:
Since there's no implementation, will it be difficult to add driver code? @chidiwilliams

Reply:
Yeah, it will. That's why I couldn't add driver code here.